bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -385,7 +385,7 @@ mod test_etc_hosts_entries {
 /// character in user-facing identifiers. It stores the original form and makes it accessible
 /// through standard traits. Its purpose is to validate input for identifiers like container names
 /// that might be used to create files/directories.
-#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
```

Comment:
Why did this need `Ord` and `PartialOrd`?

Reply:
We will be running BootstrapCommands in lexicographical order of their names. This allows sorting the keys in the Bootstrap Commands settings HashMap.

Reply:
> This allows sorting the keys in the Bootstrap Commands settings HashMap.

I don't love the implicit coupling with an agent's implementation, which as a rule should be mediated by a config file rather than directly reusing these types. But I'll review https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/62 and see if it seems OK.

Reply:
This is the relevant line - https://github.com/piyush-jena/bottlerocket-core-kit/blob/d0fcf039f2d5d82755ba885b4f76cdf34de7b658/sources/api/bootstrap-commands/src/main.rs#L204

Reply:
An escape hatch here is that any struct implemented via the `string_impls_for!` macro supports `ToString::to_string`.

On the implementation side, you could stringify these values prior to sorting them.